aptos-move/mvhashmap/src/versioned_data.rs

```diff
@@ -361,6 +361,37 @@ impl<K: Hash + Clone + Debug + Eq, V: TransactionWrite> VersionedData<K, V> {
         }));
     }
 
+    /// Versioned write of metadata at a given resource group key (and version).
+    pub fn write_metadata(
+        &self,
+        key: K,
+        txn_idx: TxnIndex,
+        incarnation: Incarnation,
+        data: V,
+    ) -> bool {
+        let arc_data = Arc::new(data);
+
+        let mut v = self.values.entry(key).or_default();
+        let prev_entry = v.versioned_map.insert(
+            ShiftedTxnIndex::new(txn_idx),
+            CachePadded::new(Entry::new_write_from(
+                incarnation,
+                ValueWithLayout::Exchanged(arc_data.clone(), None),
+            )),
+        );
+
+        // Changes versioned metadata that was stored.
+        prev_entry.map_or(true, |entry| -> bool {
+            if let EntryCell::Write(_, existing_v) = &entry.cell {
+                !existing_v
+                    .extract_value_no_layout()
+                    .eq_ignoring_bytes(&*arc_data)
```

Comment:
nit: I think `as_ref()` can be used instead of `&*`?

Reply:
got rid of eq_ignoring_bytes